File: Frontend/src/events/converters/parties/digimons/in-combat.converter.ts
import type { InCombatDTO } from "@/events/dto/parties/digimons/in-combat.dto";
import type { InCombat } from "@/models/party/digimon/in-combat";
import { VitalConverter } from "./vital.converter";

export class InCombatConverter {
  public static convert(newInCombatDto: InCombatDTO | null): InCombat {
    return {
      condition: newInCombatDto?.condition ?? 0,
      hp: VitalConverter.convert(newInCombatDto?.hp ?? null),
      mp: VitalConverter.convert(newInCombatDto?.mp ?? null),
    };
  }
}

File: Backend/Events/Converters/Parties/Digimons/InCombatConverter.cs
using Backend.Domain.Models.Parties.Digimons;
using Backend.Events.DTO.Parties.Digimons;

namespace Backend.Events.Converters.Parties.Digimons;

public static class InCombatConverter
{
    public static InCombatDTO ToDTO(InCombat digimonInCombat) => new()
    {
        Condition = digimonInCombat.Condition,
        Strength = digimonInCombat.Strength,
        Defense = digimonInCombat.Defense,
        Speed = digimonInCombat.Speed,
        HP = VitalConverter.ToDTO(digimonInCombat.HP),
        MP = VitalConverter.ToDTO(digimonInCombat.MP)
    };
}
